src/network-analytics/azext_network_analytics/aaz/latest/network_analytics/data_product/_ingest.py

```diff
@@ -0,0 +1,168 @@
+
+# pylint: skip-file
+# flake8: noqa
+
+import os
+import re
+
+from azure.cli.command_modules.keyvault._client_factory import data_plane_azure_keyvault_secret_client
+from azure.cli.core.aaz import *
+from azure.cli.core.azclierror import InvalidArgumentValueError, UnauthorizedError
+from azure.core.exceptions import ClientAuthenticationError
+from azure.cli.core.profiles import ResourceType, get_sdk
+from azure.cli.core.commands.client_factory import get_mgmt_service_client
+from datetime import date
+
+@register_command(
+    "network-analytics data-product ingest",
+)
+class Ingest(AAZCommand):
+    """Ingest sample data file into data product.
+
+    :example: Ingest sample data file with all parameters
+        az network-analytics data-product ingest --data-product-name <dpname> --resource-group <rg> --data-type <datatype> --srcdir <filepath>
+    """
+
+    def _handler(self, command_args):
+        super()._handler(command_args)
+        self._execute_operations()
+        return self._output()
+
+    _args_schema = None
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        if cls._args_schema is not None:
+            return cls._args_schema
+        cls._args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        # define Arg Group ""
+
+        _args_schema = cls._args_schema
+        _args_schema.data_product_name = AAZStrArg(
+            options=["--data-product-name"],
+            help="The data product resource name",
+            required=True,
+            id_part="name",
+            fmt=AAZStrArgFormat(
+                pattern="^[a-z][a-z0-9]*$",
+                max_length=63,
+                min_length=3,
+            ),
+        )
+        _args_schema.resource_group = AAZResourceGroupNameArg(
+            required=True,
+        )
+
+        # define Arg Group "Body"
+
+        _args_schema = cls._args_schema
+        _args_schema.data_type = AAZStrArg(
+            options=["--data-type"],
+            arg_group="Body",
+            help="Data Type.",
+            required=True,
+        )
+        _args_schema.source = AAZStrArg(
+            options=["--srcdir"],
+            arg_group="Body",
+            help="Source directory path.",
+            required=True,
+        )
+        return cls._args_schema
+    
+    def _execute_operations(self):
+        self.pre_operations()
+        self.DataProductsIngest(ctx=self.ctx)()
+        self.post_operations()
+
+    @register_callback
+    def pre_operations(self):
+        pass
+
+    @register_callback
+    def post_operations(self):
+        pass
+
+    def _output(self, *args, **kwargs):
+        result = self.deserialize_output(self.ctx.vars.instance, client_flatten=True)
+        return result
+    
+    class DataProductsIngest(object):
+        DATA_PRODUCT_ARM_ID = "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.NetworkAnalytics/dataProducts/{}"
+        API_VERSION = "2023-11-15"
```

Comment:
I first decided to hard code the API version as all the other commands in our extension have it hard coded, however, I have learnt since that the code is autogenerated and is overwritten if any changes occur. You are right, I have changed it so that it pulls the latest version